### PR TITLE
🚧 removed all deprecations 🚧

### PR DESCRIPTION
**Pull request by Climbatize**
# Description
Removed all deprecated methods, variables and unit tests from the SDK
🚧🚧🚧 DON'T MERGE THIS PR BEFORE MAJOR RELEASE IS APPROVED 🚧🚧🚧

## Links
[MINI-2647](https://jira.rakuten-it.com/jira/browse/MINI-2647)

# Checklist
- [X] I have read the [contributing guidelines](CONTRIBUTING.md).
- [X] I have completed the [SDK Development Learning Path](CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [X] I wrote/updated tests for new/changed code
- [X] I removed all sensitive data before every commit, including API endpoints and keys
- [X] I ran `fastlane ci` without errors

### CODE REVIEW

File: MiniApp/Classes/core/Protocols/MiniAppMessageDelegate.swift [hunk 1]
```diff
@@ -14,26 +11,12 @@ public protocol MiniAppMessageDelegate: MiniAppUserInfoDelegate, MiniAppShareCon
     /// result (allowed/denied) should be returned.
     func requestDevicePermission(permissionType: MiniAppDevicePermissionType, completionHandler: @escaping (Result<MASDKPermissionResponse, MASDKPermissionError>) -> Void)
 
-    /// Interface that should be implemented in the host app that handles the code to request any permission and the
-    /// result (allowed/denied) should be returned.
-    @available(*, deprecated,
-        message: "Since version 2.8.0, this method is deprecated",
-        renamed: "requestDevicePermission(permissionType:completionHandler:)")
-    func requestPermission(permissionType: MiniAppPermissionType, completionHandler: @escaping (Result<MASDKPermissionResponse, MASDKPermissionError>) -> Void)
```
Comment (corycaywood) [inline MiniApp/Classes/core/Protocols/MiniAppMessageDelegate.swift:22]: I think there was also a default implementation for this which needs to be refactored/removed. See this PR: 
https://github.com/rakutentech/ios-miniapp/pull/200/files#diff-607bf474e1963f1a8e4c576602075f15b9927f34ccb84af201e017af01bfa0daR62

File: MiniApp/Classes/core/Protocols/MiniAppMessageDelegate.swift [hunk 2]
```diff
@@ -14,26 +11,12 @@ public protocol MiniAppMessageDelegate: MiniAppUserInfoDelegate, MiniAppShareCon
...
     func requestCustomPermissions(permissions: [MASDKCustomPermissionModel],
                                   miniAppTitle: String,
                                   completionHandler: @escaping (Result<[MASDKCustomPermissionModel], MASDKCustomPermissionError>) -> Void)
 
-    /// Interface that should be implemented in the host app to handle the Custom Permissions.
-    /// Host app is responsible to display the alert/dialog with the [MiniAppCustomPermissionType] permissions to the user and the result should be returned back to the SDK
-    @available(*, deprecated,
-        message: "Since version 2.3.0, this method is deprecated",
-        renamed: "requestCustomPermissions(permissions:miniAppTitle:completionHandler:)")
-    func requestCustomPermissions(permissions: [MASDKCustomPermissionModel],
```
Comment (corycaywood) [inline MiniApp/Classes/core/Protocols/MiniAppMessageDelegate.swift:35]: There was also a default implementation for this one which needs to be refactored/removed. See below in this file.

**Review by corycaywood — COMMENTED** [review]
Looking good, but please also update the changelog with the removed public interfaces.

File: MiniApp/Classes/core/Protocols/MiniAppMessageDelegate.swift (hunk 1 above)
Comment (Climbatize) [reply MiniApp/Classes/core/Protocols/MiniAppMessageDelegate.swift:22]: oh, I'm sorry, that's really strange they did not appear in my global search and XCode did not pop a warning :-/
I'll fix this 🤔

**Review by ts-leojoseph-r — COMMENTED** [review]
Overall looks great. Please check my comment. Also, do you think we can remove/refactor
 `public typealias MiniAppDisplayProtocol = MiniAppDisplayDelegate` as well ?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,9 +1,25 @@
 ## CHANGELOG
 
-### 2.x.x
+### 3.0.0
 
 **SDK**
 
+- **Refactor** Removed all deprecated and variables methods from v2:
```
Comment (corycaywood) [inline CHANGELOG.md:7]: Maybe this should be `**Removed**` instead?

**Review by corycaywood — APPROVED** [review]
LGTM 👍

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,9 +1,25 @@
...
+  - `MiniApp` :
+    - `list(completionHandler:)` now takes a `(Result<[MiniAppInfo], MASDKError>) -> Void` in place of a `(Result<[MiniAppInfo], Error>) -> Void`
+    - `info(miniAppId:completionHandler:)` now takes a `(Result<MiniAppInfo, MASDKError>) -> Void` in place of a `(Result<MiniAppInfo, Error>) -> Void`
+    - `create(appId:completionHandler:)` now takes a `(Result<MiniAppDisplayProtocol, MASDKError>) -> Void` in place of a `(Result<MiniAppDisplayProtocol, Error>) -> Void`
+  - `MiniAppSdkConfig`:
+    - `testMode` has been removed. Use `previewMode` instead
+    - `appId` has been removed. Use `projectId`
```
[the remark this answers is not in the document]
Comment (Climbatize) [reply CHANGELOG.md:14]: yes but they are part of our Internal variables

File: CHANGELOG.md (hunk 3 above)
Comment (Climbatize) [reply CHANGELOG.md:7]: sure

**Review by ts-leojoseph-r — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**